transport.go

```diff
@@ -21,15 +21,17 @@ func ConnectionFailure(*http.Request) (*http.Response, error) {
 
 // NewMockTransport creates a new *MockTransport with no responders.
 func NewMockTransport() *MockTransport {
-	return &MockTransport{make(map[string]Responder), nil}
+	return &MockTransport{make(map[string]Responder), nil, make(map[string]int), 0}
 }
 
 // MockTransport implements http.RoundTripper, which fulfills single http requests issued by
 // an http.Client.  This implementation doesn't actually make the call, instead deferring to
 // the registered list of responders.
 type MockTransport struct {
-	responders  map[string]Responder
-	noResponder Responder
+	responders     map[string]Responder
+	noResponder    Responder
+	callCountInfo  map[string]int
+	totalCallCount int
```

Comment:
Sorry @SchumacherFM , I didn't know you were keeping an eye on this as well. There's no mutex locks in the library as it stands right now, and the library is most definitely not thread-safe for parallel test for now, so I'm not sure how much benefit there would be to add sync capabilities to this specific piece of code for now? I think it would be good to do it *in general*, but then adding and removing responders should probably also be locked as well.